### PR TITLE
fix: multiple requests locale

### PR DESCRIPTION
**Pull request by Marc-Roig**
### What does it do?

List View page was doing individual API requests for every entry.

i18n bulkActions and i18n localizations Cell was making use of `useDocument` hook, which was triggering those requests.
This PR proposes a fix to prevent those calls.

Before this PR, this was the network tab:
image

After this PR:
image

### Why is it needed?

Describe the issue you are solving.

### How to test it?

- Have more than 1 locale configured 
- Access ListView of a Content type with i18n enabled
- See individual api requests made for each entry

### Related issue(s)/PR(s)

Let us know if this is related to any issue/pull request

### CODE REVIEW

File: packages/plugins/i18n/admin/src/components/LocaleListCell.tsx [hunk 1]
```diff
@@ -9,37 +9,24 @@ import { useGetLocalesQuery } from '../services/locales';
 interface LocaleListCellProps {
   documentId: string;
   collectionType: string;
+  localizations: { locale: string }[];
   locale: string;
   model: string;
```
Comment (remidej) [inline packages/plugins/i18n/admin/src/components/LocaleListCell.tsx:14]: Should you remove those other unused props from this type? Or will they be passed upstream even if we don't use them?

**Review by remidej — COMMENTED** [review]
Less code, fewer requests, love it. I guess it was like this because we had initially removed the localizations array in v5?

haven't tested because it's a draft but it looks good to me

File: packages/plugins/i18n/admin/src/components/LocaleListCell.tsx (hunk 1 above)
Comment (Marc-Roig) [reply packages/plugins/i18n/admin/src/components/LocaleListCell.tsx]: absolutely!

**Comment by Marc-Roig** [comment]
@remidej I have uncovered a bigger issue in the content manager, I will probably need another day to fix it

File: packages/core/content-manager/server/src/services/utils/__tests__/validatable-fields-populate.test.ts [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+      const result = getPopulateForValidation(uid as any);
+
+      expect(result).toEqual({
+        populate: {
+          componentAttrName: {
+            fields: ['subfield1'], // Only required fields in the component
+          },
+        },
+      });
+    });
```
Comment (Marc-Roig) [inline packages/core/content-manager/server/src/services/utils/__tests__/validatable-fields-populate.test.ts:84]: Sooo, I am trying to create a function to select fields inside components, but it turns out there is a bug that prevents this. Until we fix that this PR will be blocked 😢 cc @remidej @markkaylor 

We use those fields populated inside the "localizations" attribute when bulk publishing locales

Comment (remidej) [reply packages/core/content-manager/server/src/services/utils/__tests__/validatable-fields-populate.test.ts:84]: Hey Marc. So to merge your improvement we'd also need to update the bulk publish functionality? What's the bug exactly?

Comment (Marc-Roig) [reply packages/core/content-manager/server/src/services/utils/__tests__/validatable-fields-populate.test.ts:84]: 1st issue that I am trying to fix: 
- the bulk publish UI needs required (and others) fields available inside `localizations` to know if a locale is publisheable or not

2nd bug I found while doing the 1st:
- we can not select fields inside component, I am not sure how we made it work in the first place (cc @jhoward1994 , maybe you can give us some hints to unblock this)
e.g of whats not possible:
```ts
populate: {
  compo: {
    fields: ['name'] // This fails
  },
}
```

Comment (Marc-Roig) [reply packages/core/content-manager/server/src/services/utils/__tests__/validatable-fields-populate.test.ts:84]: I am going to take a look at the failing API tests too

Comment (Marc-Roig) [reply packages/core/content-manager/server/src/services/utils/__tests__/validatable-fields-populate.test.ts:84]: For @jhoward1994 

the current getValidatableFieldsPopulate() function doesnt give a right output atm, for the address content type:

image

It tries to populate non populatable fields (city, postal_code), which makes the document service validation fail.

I have tried to propose a different approach, but fields can not be selected in components at the moment

**Comment by Marc-Roig** [comment]
I opened another PR with an additional fix https://github.com/strapi/strapi/pull/22330
Didnt want to polute this pr even more

File: packages/core/content-manager/server/src/services/document-metadata.ts [hunk 3]
```diff
@@ -227,13 +205,12 @@ export default ({ strapi }: { strapi: Core.Strapi }) => ({
     version: DocumentVersion,
     { availableLocales = true, availableStatus = true }: GetMetadataOptions = {}
   ) {
-    // TODO: Ignore publishedAt if availableStatus=false, and ignore locale if
+    // TODO: Ignore publishedAt aif availableStatus=false, and ignore locale if
```
Comment (jhoward1994) [inline packages/core/content-manager/server/src/services/document-metadata.ts:208]:
```suggestion
    // TODO: Ignore publishedAt if availableStatus=false, and ignore locale if
```

**Comment by remidej** [comment]
@Marc-Roig your screenshot show 1 less mapEntriesToReleases call, does your change also have impact on that?

**Review by remidej — APPROVED** [review]
Tested (the cross locale validations too) and looks all good! Thanks for walking me through it

**Pull request merged**